crates/fiber-lib/src/fiber/channel.rs

```diff
@@ -1196,8 +1228,14 @@ where
 
         let (tlc_info, remove_reason) = state.remove_tlc_with_reason(tlc_id)?;
 
-        if matches!(remove_reason, RemoveTlcReason::RemoveTlcFulfill(_)) {
+        // We should only update invoice status to paid if the peer is paying us (i.e., the TLC is received).
+        // For the TLC sent by ourselves (we're paying the peer), we should leave any invoice as it is.
+        // The reason for this is that it is possible that we're paying another peer by sending a TLC and
+        // we have a local invoice with the same payment hash. See the test
+        // test_store_update_subscription_mock_cross_chain_payment for a concrete example.
+        if tlc_id.is_received() && matches!(remove_reason, RemoveTlcReason::RemoveTlcFulfill(_)) {
```

Comment:
This check makes the `test_network_send_payment_normal_keysend_workflow` failed.

Reply:
I don't think we should remove this check, as explained in the comments, this condition is used to avoid situations like the test test_store_update_subscription_mock_cross_chain_payment. It is better to check the concrete reason why test_network_send_payment_normal_keysend_workflow failed and update the conditions if necessary.

Reply:
The cause that the 4 new cch test cases fail is that preimage has been cleaned when fetching the payment updated event in the StorePublisher actor asynchrouniously. I have refactored #736 to emit the store update event synchrouniously.